moveit_experimental/jog_arm/src/jog_arm/jog_arm_server.cpp

```diff
@@ -0,0 +1,1254 @@
+//
+// BSD 3-Clause License
+//
+// Copyright (c) 2018, Los Alamos National Security, LLC
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of source code must retain the above copyright notice, this
+//   list of conditions and the following disclaimer.
+//
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+//
+// * Neither the name of the copyright holder nor the names of its
+//   contributors may be used to endorse or promote products derived from
+//   this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+///////////////////////////////////////////////////////////////////////////////
+
+// Server node for arm jogging with MoveIt.
+
+#include <jog_arm/jog_arm_server.h>
+#include <memory>
+
+// Initialize these static struct to hold ROS parameters.
+// They must be static because they are used as arguments in thread creation.
+jog_arm::JogArmParameters jog_arm::JogROSInterface::ros_parameters_;
+jog_arm::jog_arm_shared jog_arm::JogROSInterface::shared_variables_;
+std::unique_ptr<robot_model_loader::RobotModelLoader> jog_arm::JogROSInterface::model_loader_ptr_ = NULL;
+
+/////////////////////////////////////////////////////////////////////////////////
+// JogROSInterface handles ROS subscriptions and instantiates the worker
+// threads.
+// One worker thread does the jogging calculations.
+// Another worker thread does collision checking.
+/////////////////////////////////////////////////////////////////////////////////
+
+static const char* const NODE_NAME = "jog_arm_server";
+static const int GAZEBO_REDUNTANT_MESSAGE_COUNT = 30;
+
+// MAIN
+int main(int argc, char** argv)
+{
+  ros::init(argc, argv, NODE_NAME);
+
+  jog_arm::JogROSInterface ros_interface;
+
+  return 0;
+}
+
+namespace jog_arm
+{
+// Constructor for the main ROS interface node
+JogROSInterface::JogROSInterface()
+{
+  ros::NodeHandle n;
```

Comment:
avoid one variable names except in very obvious cases like time t... ``nh`` is common